backend/amd64/emit.ml

```diff
@@ -43,11 +43,49 @@ module D = Asm_targets.Asm_directives
 module S = Asm_targets.Asm_symbol
 module L = Asm_targets.Asm_label
 
+(* Name of current function *)
+let function_name = ref ""
+
+(* Keep the name of the current block section to get back to it after emitting
+   data. *)
+let current_basic_block_section = ref ""
+
 module I = struct
   include I
 
-  let simd simd args =
-    Arch.Extension.require_instruction simd;
+  exception Extension_disabled of string
+
+  let () =
+    Printexc.register_printer (function
+      | Extension_disabled msg -> Some ("extension disabled: " ^ msg)
+      | _ -> None)
+
+  let require_vec256 () =
+    if not (Arch.Extension.enabled_vec256 ())
+    then
+      raise
+        (Extension_disabled
+           (Printf.sprintf
+              "found 256-bit register while emitting %s, but AVX is not enabled."
```

Comment:
(for consistency, that should be "whilst"...
not that I could explain the difference...)